lua/telescope/_extensions/init.lua

```diff
@@ -54,7 +55,10 @@ extensions.register = function(mod)
 end
 
 extensions.load = function(name)
-  return extensions.manager[name]
+  local ext = load_extension(name)
```

Comment:
Isnt the extension not inside `extensions.manager` do we need to do load twice? Am i missing something

Reply:
This is a compromise to maintain the current interface. We now load the extension upon access and with `telescope.load_extension $EXT` here, such that.

```lua

-- first access triggers setup without config passed, also status quo
local fb_actions = require "telescope".extensions.file_browser.actions

local telescope = require "telescope"
telescope.setup {
  extensions = {
    -- we only now set up the extension
    file_browser = { ... }
  }
}
-- now we apply the new settings and have to load extension again
telescope.load_extension "file_browser"
```

I would probably only "load" (ie applying settings) the extension in

```lua
telescope.setup {
  extensions = {
    -- we only now set up the extension
    file_browser = { ... }
  }
}
```
and maybe deprecate `load_extension`? But might not work easily with extensions like `telescope-fzf-native` (opting out then annoying I suppose?).

Compromise: just load setup on `load_extension` then?

Reply:
Oh i was just saying. Cant you do 
```lua
local ext = extensions.manager[name]
if ext.setup then
  ext.setup(extensions._config[name] or {}, require("telescope.config").values)
end
```

rather than having another local function ?

Reply:
No, `extensions.manager[name]` gives you the `exports` in

```lua
extension = {
  setup = { ... },
  exports = { ... },
}
```

Thus, `setup` is not part of `ext` in your example code.

Reply:
Oh right. Yeah i was missing something :laughing: You can merge this PR if its done.